### DISCUSSION TITLE
Debugging in worker mode?

### DISCUSSION THREAD

**Discussion by wizhippo** [php/frankenphp#234] [General]
I have xdebug working fine in non workermode, but if I enable worker mode it not longer works.

Is there any reason why worker mode is not supporting xdebug?

**Comment by f3l1x** [comment]
Same for me.

**Comment by dunghv** [comment]
same for me

**Comment by abda11ah** [comment]
"listen to xdebug" works fine in vscode or Windsurf using PHP Debug v1.35.0 vscode extension.

But for the docker version i had to use this config in .vscode/launch.json :

```
        {
            "name": "Listen for Xdebug",
            "type": "php",
            "request": "launch",
            "port": 9003,
            "pathMappings": {
                "/app/html": "${workspaceFolder}/html",
            }
        }

```

and in php.ini 

```
[Xdebug]
xdebug.mode=debug
xdebug.start_with_request=yes
xdebug.idekey="vsc"
xdebug.client_host=172.17.0.1
xdebug.client_port=9003
xdebug.discover_client_host=true
xdebug.log_level=0
```

172.17.0.1 is the docker host address

**Comment by withinboredom** [comment]
For worker requests, use [xedebug_connect_to_client()](https://xdebug.org/docs/all_functions#xdebug_connect_to_client)
